### PR TITLE
🎟️ Add internal ticketer type part 1

### PR DESCRIPTION
**Pull request by rowanseymour**
Automatically added like a system group to all orgs - hidden to all but beta orgs for now

To avoid race condition, data migration to add to existing orgs in later PR.

Mailroom changes in https://github.com/nyaruka/mailroom/pull/387

### CODE REVIEW

File: temba/tickets/models.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ class Ticketer(SmartModel):
     config = JSONField()
 
     @classmethod
-    def create(cls, org, user, ticketer_type, name, config):
+    def create(cls, org, user, ticketer_type: str, name: str, config: dict):
```
Comment (rowanseymour) [inline temba/tickets/models.py:85]: side note: in the past I've tended to use `typing.Dict` but in latest Python 3.9 `dict` now supports generic arguments making `Dict` kinda redundant

**Comment by nicpottier** [comment]
So is the vision that every org has our internal ticketer but then can opt out? And that means all orgs have ticketing by default? Just wondering if ticketing is instead something orgs should opt into. (along with the flow actions etc..)

**Comment by rowanseymour** [comment]
I guess I can't think why an org would want/need to opt-out of ticketing, and I'm not sure there's a good reason to make users find a button somewhere that says "Enable Ticketing".

But it's all beta-gated for now and people feel strongly we could add a UI level toggle - and still let every org have a system ticketer in the database the way they have system groups and fields.

**Comment by nicpottier** [comment]
Ok. Guess I feel that it is maybe a bit premature to create a ticketer for every org when we consider this beta. But if you feel strongly that's the right thing then I'm fine with it.

**Review by nicpottier — APPROVED** [review]

**Comment by rowanseymour** [comment]
There are two different questions here. I don't feel that strongly about either so make a case if you think it should be different.

1. **Long term**, what should users have to do to access ticketing. When do things like agent management become visible? Currently if you delete all your connected ticketers, you lose access access to tickets in the UI. But it wouldn't make sense to lose access to parts of the UI like managing your Agent users (we can't hide org logins).

So my inclination here is that ticketing is a first class thing, and always available. Not convinced there would be a good argument for making them click a button to enable it.

2. **Short term**, how is this all gated.

This PR creates a ticketer for every org but it's got no config and it's only visible to beta users in flows so it doesn't feel like we're committing ourselves to very much.

Having said that it feels a little odd that a beta user can add it to a flow but if a non-beta user views that same flow it's going to be a missing dependency. So I don't hate temporarily making beta users click a button to create it, and once it exists it works for everyone in that org.

**Comment by nicpottier** [comment]
My preference would be that it is opt-in for the time being especially since it is all beta. Having ticket actions appear on everybody's account right now doesn't seem like the right thing yet.

**Comment by rowanseymour** [comment]
> My preference would be that it is opt-in for the time being especially since it is all beta. Having ticket actions appear on everybody's account right now doesn't seem like the right thing yet.

But you understand that that doesn't happen in this PR?

**Comment by nicpottier** [comment]
It would for new accounts wouldn't it?

**Comment by nicpottier** [comment]
Oh I see the API filter by beta. Ok. Still think it's weird, but as I said if you really want it go for it!

**Comment by rowanseymour** [comment]
That's not a helpful PR comment

**Comment by rowanseymour** [comment]
Updated so internal ticketers created explicitly for now

Captura de Pantalla 2020-12-09 a la(s) 17 19 14

**Comment by nicpottier** [comment]
For beta I think that's fine.

**Pull request merged**